openprescribing/frontend/management/commands/scrape_numbers.py

```diff
@@ -0,0 +1,243 @@
+deletes old records of numbers.
+'''
+
+import json
+import os
+import re
+from datetime import datetime
+from glob import glob
+from time import sleep
+
+from django.conf import settings
+from django.core.management import BaseCommand
+from django.core.urlresolvers import get_resolver
+
+from bs4 import BeautifulSoup
+from selenium import webdriver
+
+from openprescribing.slack import notify_slack
+from openprescribing.utils import mkdir_p
+from pipeline.runner import in_progress as import_in_progress
+
+
+class Command(BaseCommand):
+    help = __doc__
+
+    def handle(self, *args, **options):
+        if import_in_progress():
+            notify_slack('Not checking numbers: import in progress')
+            return
+
+        previous_log_path = get_previous_log_path()
+
+        timestamp = datetime.now().strftime('%Y%m%d%H%M%S')
+        log_path = os.path.join(settings.CHECK_NUMBERS_BASE_PATH, timestamp)
+        mkdir_p(log_path)
+
+        with webdriver.Firefox() as browser:
+            numbers = {
+                name: {
+                    'path': path,
+                    'numbers': extract_numbers(browser, path, name, log_path),
+                }
+                for name, path in get_paths_to_scrape()
+            } 
+
+        write_numbers(numbers, log_path)
+
+        if previous_log_path is None:
+            msg = 'Not checking numbers: this is the first deploy since last import'
+            notify_slack(msg)
+            return
+
+        previous_numbers = load_previous_numbers(previous_log_path)
+
+        differences = compare_numbers(previous_numbers, numbers)
+
+        if differences:
+            msg = 'The following pages have changed:\n\n'
+            msg += '\n'.join(differences)
+            msg += '\n\nNext step: compare {} and {}'.format(previous_log_path, log_path)
+            notify_slack(msg)
+
+
+def get_paths_to_scrape():
+    '''Yield paths that should be scraped.
+
+    We're interested in a sample of all pages for organisations.  Rather than
+    specify pages we're interested in, we ignore pages we're not interested in.
+    '''
+
+    # Don't scrape URLs beginning with these prefixes.
+    prefixes_to_ignore = [
+        'accounts',
+        'admin',
+        'api',
+        'bnf',
+        'bookmarks',
+        'chemical',
+        'dmd',
+        'docs',
+    ]
+
+    # get_resolver().reverse_dict is a dict that maps view names or view
+    # functions to a data structure that describes how requests should be
+    # dispatched.
+    for k, v in get_resolver().reverse_dict.items():
+
+        # Ignore records where the key is a view function.
+        if not isinstance(k, str):
+            continue
+
+        name = k
+        pattern = v[0][0][0]
+        keys = v[0][0][1]
+
+        # Ignore records starting with prefixes we're not interested in.
+        if any(pattern.startswith(prefix) for prefix in prefixes_to_ignore):
+            continue
+
+        # Ignore any URLs that 
```

Comment:
Typo / incomplete sentence. And it would be useful because the purpose isn't obvious without extra code-reading elsewhere, and domain knowledge about how/why all-england is special.